osgar/tools/lidarview.py

```diff
@@ -58,6 +58,10 @@ def scans_gen(logfile, lidar_name=None, poses_name=None, camera_name=None):
                 if lidar_id is None and camera_id is None:
                     yield timestamp, pose, scan, image
 
+        # generate last message for ever ...
+        while True:
+            yield timestamp, pose, scan, image
```

Comment:
Huh. I understand that this is the simplest way how to somehow "stop at the end" but... it is not actually stopping. It is just redrawing the same thing at 10Hz.

Reply:
Yes, it was the easiest workaround, but I do not like it any more. Also the queue full of timer events makes the viewer for 1GB logfiles unresponsive ... OK, one more revision at least.